### PR TITLE
PS-8854: Convert percona-udf to a component with function auto-registration

### PR DESCRIPTION
**Pull request by inikep**

### CODE REVIEW

File: components/percona-udf/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+MYSQL_ADD_COMPONENT(percona_udf
+  percona_udf.cc
+  fnv1a_udf.cc
+  fnv_udf.cc
+  murmur_udf.cc
+  MODULE_ONLY)
+
```
Comment (percona-ysorokin) [inline components/percona-udf/CMakeLists.txt:23]: Also consider making all symbols hidden by default
```
set_target_properties(component_masking_functions PROPERTIES
  CXX_VISIBILITY_PRESET hidden
  VISIBILITY_INLINES_HIDDEN YES
)
```
Probably for backward compatibility (for those who still want to use this .so as a source of user-defined functions via INSTALL FUNCTION), we can also make
* fnv_64
* fnv_64_init
* fnva_64
* fnva_64_init
* murmur_hash
* murmur_hash_init
explicitly exported (`visibility=default`)

**Review by percona-ysorokin — CHANGES_REQUESTED** [review]
Please also wait for the Masking Functions component PR - it has function auto-registration code put in common headers.

File: components/percona-udf/CMakeLists.txt (hunk 1 above)
Comment (inikep) [reply components/percona-udf/CMakeLists.txt:23]: 8.1.0 sets `-fvisibility=hidden` for components by default:
https://github.com/inikep/percona-server/blob/internal-8.1.0-linear/cmake/component.cmake#L115

**Comment by inikep** [comment]
I added backward compatibility for
```
CREATE FUNCTION fnv1a_64 RETURNS INTEGER SONAME 'component_percona_udf.so';
CREATE FUNCTION fnv_64 RETURNS INTEGER SONAME 'component_percona_udf.so';
CREATE FUNCTION murmur_hash RETURNS INTEGER SONAME 'component_percona_udf.so';
```
and related MTR checks using `__attribute__ ((visibility ("default")))`.
 
I'm waiting for auto-registration code.

**Pull request closed without merging**